app_dart/lib/src/request_handlers/override_github_build_status.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/service/github_service.dart';
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../model/appengine/github_build_status_update.dart';
+import '../model/appengine/github_tree_status_override.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../request_handling/exceptions.dart';
+import '../service/datastore.dart';
+
+@immutable
+class OverrideGitHubBuildStatus extends ApiRequestHandler<Body> {
+  const OverrideGitHubBuildStatus(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting DatastoreServiceProvider datastoreProvider,
+    @visibleForTesting LoggingProvider loggingProvider,
+  })  : datastoreProvider = datastoreProvider ?? DatastoreService.defaultProvider,
+        loggingProvider = loggingProvider ?? Providers.serviceScopeLogger,
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+  final LoggingProvider loggingProvider;
+
+  static const String repositoryKeyName = 'repository';
+  static const String closedKeyName = 'closed';
+  static const String reasonKeyName = 'reason';
+  static const String forceKeyName = 'force';
+
+  @override
+  Future<Body> get() async {
+    final DatastoreService datastore = datastoreProvider(config.db);
+    final List<GithubTreeStatusOverride> overrides =
+        await datastore.db.query<GithubTreeStatusOverride>().run().toList();
+
+    return Body.forJson(overrides);
+  }
+
+  @override
+  Future<Body> put() async {
```

Comment:
Is it possible to describe the api in the top of this file? it will be useful for people not very familiar with cocoon.

Reply:
Sure. Are you looking for docs both on the class and the put method?

Reply:
Both, I assume this api will be called later from the dashboard ui but it will also be very useful to see a few examples of how to use it when reading the code.